toolboxTreppe/toolboxTreppe.ino

```diff
@@ -146,8 +141,13 @@ void powerOff() {
 }
 
 //------------------------ Matrix definitionen ----------
+const uint16_t XYsafe( const uint8_t& x, const uint8_t& y) {
+  if( x >= kMatrixWidth) return -1;
+  if( y >= kMatrixHeight) return -1;
+  return XY(x,y);
+}
 
-uint16_t XY(const uint8_t x, const uint8_t y) {
+const uint16_t XY(const uint8_t& x, const uint8_t& y) {
```

Comment:
XY could be a struct with constructor

Reply:
basically none - just style

Reply:
Hm I am still not quite sure how you mean this
I mean basically XY calculates a pin number so when I understand you right you would prefer a struct
that includes the variable pinNumber and the constructor of the struct gets called with the parameters x and y, calculates the pin number and sets the pinNumber variable to this value
Then you would have to access this value.

In comparision the current way you call the function and it returns the pin that you need at this place
I mean you only need this pin once, so I can´t really see a big advantage in using a struct even in terms of style, or is my idea on how you would implement this struct

Reply:
kanns dir einfacher am do mal erklären